attributes/default.rb

```diff
@@ -62,10 +62,14 @@
 }
 architecture_map.default = 'x86_64'
 
+# Older versions of yum embed M2Crypto with SSL that doesn't support TLS1.2
+prefix = node['platform_version'].to_i < 6 ? 'http' : 'https'
```

Comment:
After giving it some thought, I think this deserves a more accurate condition since it will run on all platforms. Could you change this to:

```ruby
yum_protocol = 'https'
if node['platform_family'] == 'rhel' && node['platform_version'].to_i < 6
  # Older versions of yum embed M2Crypto with SSL that doesn't support TLS1.2
  yum_protocol = 'http'
end
```

(I've also renamed `prefix` `yum_protocol` since that name makes more sense here IMO)